diff -ru ruby-2.0.0-p0-orig/configure ruby-2.0.0-p0/configure
--- ruby-2.0.0-p0-orig/configure	2013-02-24 00:02:47.000000000 -0800
+++ ruby-2.0.0-p0/configure	2013-03-16 22:07:32.297792000 -0700
@@ -15583,7 +15583,7 @@
 	      truncate64 ftruncate64 ftello64 fseeko fseeko64 \
 	      link symlink readlink readdir_r fsync fdatasync fchown posix_fadvise\
 	      setitimer setruid seteuid setreuid setresuid socketpair\
-	      setrgid setegid setregid setresgid issetugid pause lchown lchmod\
+	      setrgid setegid setregid setresgid pause lchown lchmod\
 	      getpgrp setpgrp getpgid setpgid initgroups getgroups setgroups\
 	      getpriority getrlimit setrlimit sysconf close getpwnam_r getgrnam_r\
 	      dlopen sigprocmask sigaction sigsetjmp _setjmp _longjmp\
@@ -16954,7 +16954,7 @@
 	pthread_getattr_np pthread_attr_get_np pthread_attr_getstack\
 	pthread_get_stackaddr_np pthread_get_stacksize_np \
 	thr_stksegment pthread_stackseg_np pthread_getthrds_np \
-	pthread_cond_init pthread_condattr_setclock pthread_condattr_init \
+	pthread_cond_init pthread_condattr_init \
         pthread_sigmask
 do :
   as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
diff -ru ruby-2.0.0-p0-orig/configure.in ruby-2.0.0-p0/configure.in
--- ruby-2.0.0-p0-orig/configure.in	2013-02-22 20:01:46.000000000 -0800
+++ ruby-2.0.0-p0/configure.in	2013-03-16 22:07:43.119792000 -0700
@@ -1572,7 +1572,7 @@
 	      truncate64 ftruncate64 ftello64 fseeko fseeko64 \
 	      link symlink readlink readdir_r fsync fdatasync fchown posix_fadvise\
 	      setitimer setruid seteuid setreuid setresuid socketpair\
-	      setrgid setegid setregid setresgid issetugid pause lchown lchmod\
+	      setrgid setegid setregid setresgid pause lchown lchmod\
 	      getpgrp setpgrp getpgid setpgid initgroups getgroups setgroups\
 	      getpriority getrlimit setrlimit sysconf close getpwnam_r getgrnam_r\
 	      dlopen sigprocmask sigaction sigsetjmp _setjmp _longjmp\
@@ -2042,7 +2042,7 @@
 	pthread_getattr_np pthread_attr_get_np pthread_attr_getstack\
 	pthread_get_stackaddr_np pthread_get_stacksize_np \
 	thr_stksegment pthread_stackseg_np pthread_getthrds_np \
-	pthread_cond_init pthread_condattr_setclock pthread_condattr_init \
+	pthread_cond_init pthread_condattr_init \
         pthread_sigmask)
     if test "${host_os}" = "nacl"; then
       ac_cv_func_pthread_attr_init=no
diff -ru ruby-2.0.0-p0-orig/ext/socket/init.c ruby-2.0.0-p0/ext/socket/init.c
--- ruby-2.0.0-p0-orig/ext/socket/init.c	2013-02-03 22:19:40.000000000 -0800
+++ ruby-2.0.0-p0/ext/socket/init.c	2013-03-16 22:23:30.225793000 -0700
@@ -486,6 +486,7 @@
     static int try_accept4 = 1;
 #endif
     if (address_len) len0 = *address_len;
+#ifdef SOCK_CLOEXEC
 #ifdef HAVE_ACCEPT4
     if (try_accept4) {
         ret = accept4(socket, address, address_len, SOCK_CLOEXEC);
@@ -502,6 +503,7 @@
         try_accept4 = 0;
     }
 #endif
+#endif
     ret = accept(socket, address, address_len);
     if (ret == -1) return -1;
     if (address_len && len0 < *address_len) *address_len = len0;
