diff -ru gitlab-orig/vendor/bundle/ruby/2.0.0/gems/grit-2.5.0/lib/grit/git-ruby/internal/pack.rb gitlab/vendor/bundle/ruby/2.0.0/gems/grit-2.5.0/lib/grit/git-ruby/internal/pack.rb
--- gitlab-orig/vendor/bundle/ruby/2.0.0/gems/grit-2.5.0/lib/grit/git-ruby/internal/pack.rb	2013-03-17 20:00:35.652806000 -0700
+++ gitlab/vendor/bundle/ruby/2.0.0/gems/grit-2.5.0/lib/grit/git-ruby/internal/pack.rb	2013-03-17 20:55:11.253904151 -0700
@@ -13,8 +13,7 @@
 require 'grit/git-ruby/internal/raw_object'
 require 'grit/git-ruby/internal/file_window'
 
-PACK_SIGNATURE = "PACK"
-PACK_IDX_SIGNATURE = "\377tOc"
+PACK_IDX_SIGNATURE = [0xFF, 0x74, 0x4F, 0x63]
 
 module Grit
   module GitRuby
@@ -57,7 +56,7 @@
           end
 
           # read header
-          sig = idxfile.read(4)
+          sig = idxfile.read(4).unpack("C*")
           ver = idxfile.read(4).unpack("N")[0]
 
           if sig == PACK_IDX_SIGNATURE
diff -ru gitlab-orig/vendor/bundle/ruby/2.0.0/gems/grit-2.5.0/lib/grit/ruby1.9.rb gitlab/vendor/bundle/ruby/2.0.0/gems/grit-2.5.0/lib/grit/ruby1.9.rb
--- gitlab-orig/vendor/bundle/ruby/2.0.0/gems/grit-2.5.0/lib/grit/ruby1.9.rb	2013-03-17 20:00:35.821792000 -0700
+++ gitlab/vendor/bundle/ruby/2.0.0/gems/grit-2.5.0/lib/grit/ruby1.9.rb	2013-03-17 20:55:04.392805708 -0700
@@ -1,7 +1,7 @@
 class String
-  if ((defined? RUBY_VERSION) && (RUBY_VERSION[0..2] == "1.9"))
+  if ((defined? RUBY_VERSION) && (['1.9', '2.0'].include? RUBY_VERSION[0..2]))
     def getord(offset); self[offset].ord; end
   else
     alias :getord :[]
   end
-end
\ No newline at end of file
+end
