diff -ru gitlab-orig/vendor/bundle/ruby/2.0.0/gems/libv8-3.3.10.4/lib/libv8/build/scons/install/lib/scons-2.1.0/SCons/Environment.py gitlab/vendor/bundle/ruby/2.0.0/gems/libv8-3.3.10.4/lib/libv8/build/scons/install/lib/scons-2.1.0/SCons/Environment.py
--- gitlab-orig/vendor/bundle/ruby/2.0.0/gems/libv8-3.3.10.4/lib/libv8/build/scons/install/lib/scons-2.1.0/SCons/Environment.py	2013-03-17 20:14:15.452791985 -0700
+++ gitlab/vendor/bundle/ruby/2.0.0/gems/libv8-3.3.10.4/lib/libv8/build/scons/install/lib/scons-2.1.0/SCons/Environment.py	2013-03-17 20:09:04.021792323 -0700
@@ -406,6 +406,8 @@
             del self._dict[key]
 
     def __getitem__(self, key):
+        if key == 'os':
+            return 'freebsd'
         return self._dict[key]
 
     def __setitem__(self, key, value):
diff -ru gitlab-orig/vendor/bundle/ruby/2.0.0/gems/libv8-3.3.10.4/lib/libv8/build/v8/SConstruct gitlab/vendor/bundle/ruby/2.0.0/gems/libv8-3.3.10.4/lib/libv8/build/v8/SConstruct
--- gitlab-orig/vendor/bundle/ruby/2.0.0/gems/libv8-3.3.10.4/lib/libv8/build/v8/SConstruct	2013-03-17 10:24:12.834792000 -0700
+++ gitlab/vendor/bundle/ruby/2.0.0/gems/libv8-3.3.10.4/lib/libv8/build/v8/SConstruct	2013-03-17 20:10:48.205792115 -0700
@@ -126,7 +126,7 @@
     'os:freebsd': {
       'CPPPATH' : ['/usr/local/include'],
       'LIBPATH' : ['/usr/local/lib'],
-      'CCFLAGS':      ['-ansi'],
+      'CCFLAGS':      ['-ansi', '-fPIC'],
       'LIBS': ['execinfo']
     },
     'os:openbsd': {
@@ -298,6 +298,9 @@
                        '-Werror',
                        '-W',
                        '-Wno-unused-parameter',
+                       '-Wno-error=unused-but-set-variable',
+                       '-Wno-error=strict-overflow',
+                       '-Wno-error=narrowing',
                        '-Wnon-virtual-dtor']
     },
     'os:win32': {
@@ -1167,7 +1170,7 @@
   if env['os'] != 'linux' and env['compress_startup_data'] != 'off':
     Abort("Startup data compression is only available on Linux")
   for (name, option) in ALL_OPTIONS.iteritems():
-    if (not name in env):
+    if (name != 'os') and (not name in env):
       message = ("A value for option %s must be specified (%s)." %
           (name, ", ".join(option['values'])))
       Abort(message)
