diff -ru init.d-orig/gitlab init.d/gitlab
--- init.d-orig/gitlab	2013-04-29 01:21:50.333578000 -0700
+++ init.d/gitlab	2013-04-29 01:23:58.808531647 -0700
@@ -15,7 +15,7 @@
 ### END INIT INFO
 
 
-APP_ROOT="/home/git/gitlab"
+APP_ROOT="/var/lib/git/gitlab"
 DAEMON_OPTS="-C $APP_ROOT/config/puma.rb -e production"
 PID_PATH="$APP_ROOT/tmp/pids"
 WEB_SERVER_PID="$PID_PATH/puma.pid"
@@ -45,8 +45,8 @@
     exit 1
   else
     if [ `whoami` = root ]; then
-      sudo -u git -H bash -l -c "RAILS_ENV=production bundle exec puma $DAEMON_OPTS"
-      sudo -u git -H bash -l -c "mkdir -p $PID_PATH && $START_SIDEKIQ  > /dev/null  2>&1 &"
+      sudo -u git -H bash -l -c ". ~/.bashrc ; RAILS_ENV=production bundle exec puma $DAEMON_OPTS"
+      sudo -u git -H bash -l -c ". ~/.bashrc ; mkdir -p $PID_PATH && $START_SIDEKIQ  > /dev/null  2>&1 &"
       echo "$DESC started"
     fi
   fi
@@ -58,7 +58,7 @@
   if [ "$PID" -ne 0 -a "$STATUS" -ne 0 ]; then
     ## Program is running, stop it.
     kill -QUIT `cat $WEB_SERVER_PID`
-    sudo -u git -H bash -l -c "mkdir -p $PID_PATH && $STOP_SIDEKIQ  > /dev/null  2>&1 &"
+    sudo -u git -H bash -l -c ". ~/.bashrc ; mkdir -p $PID_PATH && $STOP_SIDEKIQ  > /dev/null  2>&1 &"
     rm "$WEB_SERVER_PID" >> /dev/null
     echo "$DESC stopped"
   else
@@ -74,9 +74,9 @@
   if [ "$PID" -ne 0 -a "$STATUS" -ne 0 ]; then
     echo "Restarting $DESC..."
     kill -USR2 `cat $WEB_SERVER_PID`
-    sudo -u git -H bash -l -c "mkdir -p $PID_PATH && $STOP_SIDEKIQ  > /dev/null  2>&1 &"
+    sudo -u git -H bash -l -c ". ~/.bashrc ; mkdir -p $PID_PATH && $STOP_SIDEKIQ  > /dev/null  2>&1 &"
     if [ `whoami` = root ]; then
-      sudo -u git -H bash -l -c "mkdir -p $PID_PATH && $START_SIDEKIQ  > /dev/null  2>&1 &"
+      sudo -u git -H bash -l -c ". ~/.bashrc ; mkdir -p $PID_PATH && $START_SIDEKIQ  > /dev/null  2>&1 &"
     fi
     echo "$DESC restarted."
   else
